src/module/rent/application/service/rent.service.ts

```diff
@@ -0,0 +1,115 @@
+import { CarService } from "../../../car/application/service/car.service";
+import { CustomerService } from "../../../customer/customer.module";
+import { Rent, StatusEnum } from "../../domain/rent.entity";
+import { CreateRentDto } from "../dto/create.rent.dto";
+import { fromModelRentToEntity } from "../mapper/fromModelRentToEntity";
+import { IRentRepository } from "../repository/rent.repository.interface";
+
+export class RentService {
+	rentRepository: IRentRepository;
+	carService: CarService;
+	customerService: CustomerService;
+	constructor(
+		rentRepository: IRentRepository,
+		carService: CarService,
+		customerSer: CustomerService
+	) {
+		this.rentRepository = rentRepository;
+		this.carService = carService;
+		this.customerService = customerSer;
+	}
+
+	async create(rent: Rent): Promise<Rent> {
+		const { carId, customerId } = rent;
+		try {
+			await this.validateTransaction(carId!, customerId!);
+			rent.totalPrice = this.setTotalPrice(
+				rent.endDate,
+				rent.startDate,
+				rent.unitPrice
+			);
+			const rentSaved = await this.rentRepository.save(rent);
+			return rentSaved;
+		} catch (error) {
+			throw error;
+		}
+	}
+
+	async checkCarRent(carId: number): Promise<false> {
+		const rent = await this.rentRepository.findByCarId(carId);
+		if (rent?.status === StatusEnum.pending)
+			throw new Error("Car is already rented");
+		return false;
+	}
+
+	async checkCustomerRent(customerId: number): Promise<false> {
+		const rent = await this.rentRepository.findByCustomerId(customerId);
+		if (rent?.status === StatusEnum.pending)
+			throw new Error("Customer has a rented");
+		return false;
+	}
+
+	async validateTransaction(
+		carId: number,
+		customerId: number
+	): Promise<void> {
+		await this.customerService.getById(customerId);
+		await this.carService.getById(carId);
+		await this.checkCarRent(carId);
+		await this.checkCustomerRent(customerId);
+	}
```

Comment:
I think you're only using these methods inside your class. If that's the case you can assign them private and skip testing them, as they're been tested by extension when you check your public methods calling them

Reply:
Sure!